oemoflex/preprocessing.py

```diff
@@ -1195,3 +1197,61 @@ def create_electricity_bev_profiles(data_raw_path, data_preprocessed_path):
         profile_df.to_csv(
             os.path.join(data_preprocessed_path, 'sequences', k + '_profile.csv')
         )
+
+
+def create_profiles(exp_path, select_components):
+
+    oemof_tabular_settings_filepath = os.path.join(module_path, 'oemof-tabular-settings.yml')
+    with open(oemof_tabular_settings_filepath, 'r') as settings_file:
+        settings = yaml.safe_load(settings_file)
+        sequences_dir = settings['sequences-dir']
+        profile_file_suffix = settings['profile-file-suffix']
+        profile_name_suffix = settings['profile-name-suffix']
+
+    mapping_filepath = os.path.join(module_path, 'mapping-input-timeseries.csv')
+
+    with open(mapping_filepath, 'r') as mapping_file:
+        map_df = pd.read_csv(mapping_file, index_col=0)
+
+    raw_path = exp_path.data_raw
+
+    preprocessed_path = exp_path.data_preprocessed
+
+    for component in select_components:
+
+        try:
+            profiles = map_df.loc[component, :]
+
+        except:
+            logging.info(f"No mapping information found for timeseries of '{component}'.")
+
+            continue
+
+        if pd.notnull(profiles.profile_name):
+            profile_name = profiles.profile_name
+
+        else:
+            profile_name = component
+
+        if pd.notnull(profiles.output_name):
+            output_filename_base = profiles.output_name
+
+        else:
+            output_filename_base = profile_name
+
+        profile_paths = os.path.join(raw_path, profiles.input_data)
+
+        logging.info(f"Creating '{profile_name}' timeseries for '{component}'.")
+
+        profile_df = combine_profiles(profile_paths, profile_name + profile_name_suffix)
+
+        if profiles.apply_function == 'normalize_year':
```

Comment:
This should be a real function.

Reply:
Could be called via globals()['apply_function'](profiles_df)

Reply:
globals() seem to dangerous to me. I'll try using a dedicated dict.